### PR TITLE
PIC-4134 put service account name in pod name - useful for namespaces…

### PR DESCRIPTION
**Pull request by markreesmoj**
… with multiple service accounts

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,6 +1,6 @@
 locals {
   # Generic configuration
-  identifier = "cloud-platform-${random_id.name.hex}"
+  identifier = "cp-${random_id.name.hex}"
```
Comment (markreesmoj) [inline main.tf:3]: Shortening the prefix to reduce the chance of hitting the 64 character limit
I wonder if the service name could replace the random_id part instead?

File: main.tf [hunk 2]
```diff
@@ -22,7 +22,7 @@ data "aws_ecr_repository" "service_pod" {
 ################################
 resource "kubernetes_deployment" "service_pod" {
   metadata {
-    name      = "${local.identifier}-service-pod"
+    name      = "${local.identifier}-${var.service_account_name}-service-pod"
```
Comment (jaskaransarkaria) [inline main.tf:25]:
```suggestion
    name      = ${var.service_account_name} == "" ? "${local.identifier}-service-pod" : "${var.service_account_name}-service-pod"
```

File: main.tf [hunk 3]
```diff
@@ -35,14 +35,14 @@ resource "kubernetes_deployment" "service_pod" {
 
     selector {
       match_labels = {
-        app = "${local.identifier}-service-pod"
+        app = "${local.identifier}-${var.service_account_name}-service-pod"
```
Comment (jaskaransarkaria) [inline main.tf:38]:
```suggestion
    name      = ${var.service_account_name} == "" ? "${local.identifier}-service-pod" : "${var.service_account_name}-service-pod"
```

File: main.tf [hunk 4]
```diff
@@ -35,14 +35,14 @@ resource "kubernetes_deployment" "service_pod" {
...
       }
     }
 
     template {
       metadata {
         labels = {
-          app = "${local.identifier}-service-pod"
+          app = "${local.identifier}-${var.service_account_name}-service-pod"
```
Comment (jaskaransarkaria) [inline main.tf:45]:
```suggestion
    name      = ${var.service_account_name} == "" ? "${local.identifier}-service-pod" : "${var.service_account_name}-service-pod"
```

File: main.tf [hunk 5]
```diff
@@ -22,7 +22,7 @@ data "aws_ecr_repository" "service_pod" {
 ################################
 resource "kubernetes_deployment" "service_pod" {
   metadata {
-    name      = "${local.identifier}-service-pod"
+    name      = ${var.service_account_name} == "" ? "${local.identifier}-service-pod" : "${var.service_account_name}-service-pod"
```
Comment (sj-williams) [inline main.tf:25]:
```suggestion
    name      = var.service_account_name == "" ? "${local.identifier}-service-pod" : "${var.service_account_name}-service-pod"
```

File: main.tf [hunk 6]
```diff
@@ -35,14 +35,14 @@ resource "kubernetes_deployment" "service_pod" {
 
     selector {
       match_labels = {
-        app = "${local.identifier}-service-pod"
+    name      = ${var.service_account_name} == "" ? "${local.identifier}-service-pod" : "${var.service_account_name}-service-pod"
```
Comment (sj-williams) [inline main.tf:38]:
```suggestion
    name      = var.service_account_name == "" ? "${local.identifier}-service-pod" : "${var.service_account_name}-service-pod"
```

File: main.tf [hunk 7]
```diff
@@ -35,14 +35,14 @@ resource "kubernetes_deployment" "service_pod" {
...
       }
     }
 
     template {
       metadata {
         labels = {
-          app = "${local.identifier}-service-pod"
+    name      = ${var.service_account_name} == "" ? "${local.identifier}-service-pod" : "${var.service_account_name}-service-pod"
```
Comment (sj-williams) [inline main.tf:45]:
```suggestion
    name      = var.service_account_name == "" ? "${local.identifier}-service-pod" : "${var.service_account_name}-service-pod"
```

Comment (sj-williams) [reply main.tf:45]: @markreesmoj just spotted that these suggested changes were using the old interpolation syntax which is no longer supported outside of string expressions (see broken unit test actions). This should work though

**Review by sj-williams — CHANGES_REQUESTED** [review]

**Review by sj-williams — APPROVED** [review]

**Pull request merged**